misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -103,11 +105,30 @@ private boolean processEvent(EventSet eventSet, Event event) {
             configureUserBreakPoints(evt.referenceType());
             eventSet.resume();
         } else if (event instanceof BreakpointEvent) {
-            StoppedEventArguments stoppedEventArguments = new StoppedEventArguments();
-            stoppedEventArguments.setReason(StoppedEventArgumentsReason.BREAKPOINT);
-            stoppedEventArguments.setThreadId(((BreakpointEvent) event).thread().uniqueID());
-            stoppedEventArguments.setAllThreadsStopped(true);
-            context.getClient().stopped(stoppedEventArguments);
+            boolean condition = false;
+            String qualifiedClassName =
+                getQualifiedClassName(context, ((BreakpointEvent) event).location().declaringType());
+            Map<Integer, SourceBreakpoint> sourceBreakpoints = this.sourceBreakpoints.get(qualifiedClassName);
+            int lineNumber = ((BreakpointEvent) event).location().lineNumber();
+
+            if (sourceBreakpoints != null && sourceBreakpoints.containsKey(lineNumber)) {
+                condition = context.getAdapter().evaluateBreakpointCondition(
+                        sourceBreakpoints.get(lineNumber).getCondition(), ((BreakpointEvent) event).thread());
+            }
+            if (sourceBreakpoints != null && sourceBreakpoints.containsKey(lineNumber)
+                && sourceBreakpoints.get(lineNumber).getCondition() != null && !condition) {
+                context.getDebuggeeVM().resume();
+                ContinuedEventArguments continuedEventArguments = new ContinuedEventArguments();
+                continuedEventArguments.setThreadId(((BreakpointEvent) event).thread().uniqueID());
+                continuedEventArguments.setAllThreadsContinued(true);
+                context.getClient().continued(continuedEventArguments);
+            } else {
+                StoppedEventArguments stoppedEventArguments = new StoppedEventArguments();
+                stoppedEventArguments.setReason(StoppedEventArgumentsReason.BREAKPOINT);
+                stoppedEventArguments.setThreadId(((BreakpointEvent) event).thread().uniqueID());
+                stoppedEventArguments.setAllThreadsStopped(true);
+                context.getClient().stopped(stoppedEventArguments);
+            }
```

Comment:
Can't we simplify the logic in here as follows?
```java
if (condition) {
   // suspend
} else {
  // continue
} 
```

Reply:
Done